single.mustache

```diff
@@ -10,12 +13,18 @@ URL: {{{PROJECTURL}}}
 {{#each SOURCES}}
 {{{.}}}
 {{/each}}
+%if 0%{?scl:1}
+BuildRequires: %{?scl_prefix_nodejs}build
+BuildRequires: %{?scl_prefix_nodejs}npm
+%else
 BuildRequires: nodejs-packaging
+%endif
 {{#each DEPENDENCIES}}
 Requires: {{{.}}}
 {{/each}}
 BuildArch: noarch
 ExclusiveArch: %{nodejs_arches} noarch
+Provides: %{?scl_prefix_nodejs}npm(%{npm_name}) = %{version}
```

Comment:
There is `/usr/lib/rpm/rh-nodejs10.prov` which should automatically do this within the SCL. `nodejs-packaging` does the same for non-SCL builds.

Reply:
This is technically wrong as I need to change to `%{?scl_prefix}` -- the question becomes does that rh-nodejs10.prov handle this ? We want this ot end up as `tfm-npm` for example and not `rh-nodejs10-npm`

Reply:
That's a good question. Technically one could say it's not directly usable in another SCL that inherits from `rh-nodejs10` so it wouldn't provide it. On the other hand, it does provide the NPM package built for that SCL. Is there some guidance from SCL how to deal with this?

Reply:
The only guidance I know of is, if your package doesn't live in the repository with the SCL itself, you should have your own SCL to clearly articulate this. (Which is what we do)